src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java

```diff
@@ -0,0 +1,80 @@
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.MovieSession;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieSessionDaoImpl implements MovieSessionDao {
+    @Override
+    public MovieSession add(MovieSession movieSession) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(movieSession);
+            transaction.commit();
+            return movieSession;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert Movie Session " + movieSession, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public Optional<MovieSession> get(Long id) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return Optional.ofNullable(session.get(MovieSession.class, id));
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get a Movie Session by id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<MovieSession> getAll() {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<MovieSession> getAllMovieSessionsQuery = session.createQuery(
+                    "FROM MovieSession", MovieSession.class);
+            return getAllMovieSessionsQuery.getResultList();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get all movie sessions", e);
+        }
+    }
+
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<MovieSession> getAllMovieSessions = session.createQuery(
+                    "FROM MovieSession m "
+                            + "JOIN FETCH m.movie "
+                            + "JOIN FETCH m.cinemaHall "
+                            + "WHERE m.movie.id = :id "
+                            + "AND " + "m.showTime BETWEEN :startOfDay AND :endOfDay",
+                    MovieSession.class);
+            getAllMovieSessions.setParameter("id", movieId);
+            getAllMovieSessions.setParameter("startOfDay", LocalDateTime.of(date, LocalTime.MIN));
+            getAllMovieSessions.setParameter("endOfDay", LocalDateTime.of(date, LocalTime.MAX));
+            return getAllMovieSessions.getResultList();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get all movie sessions", e);

```

Comment:
All movie sessions? Or movie sessions by movie id and date?

Reply:
not resolved